src/cincinnati/mod.rs

```diff
@@ -277,7 +277,13 @@ fn evaluate_deadend(node: &Node) -> Option<String> {
     if deadend_reason.is_empty() {
         deadend_reason = "(unknown reason)".to_string();
     }
-
+    if !deadend_reason.is_empty() {
```

Comment:
This can never be true, right? We set it to a default value just before this.

Reply:
Almost, this is a tautological check (condition is always true).
The root problem here is that this logic is in the wrong place; it should be pushed one level up, handling the `Option<String>`.